packages/gatsby-link/src/index.tsx

```diff
@@ -1,12 +1,48 @@
 import PropTypes from "prop-types"
 import React from "react"
-import { Link } from "@reach/router"
+import {
+  Link,
+  LinkProps,
+  LinkGetProps,
+  NavigateFn,
+  NavigateOptions,
+} from "@reach/router"
 
 import { parsePath } from "./parse-path"
 
 export { parsePath }
 
-export function withPrefix(path) {
+export interface IGatsbyLinkState {
+  IOSupported: boolean
+}
+
+// eslint-disable-next-line @typescript-eslint/interface-name-prefix
+export interface GatsbyLinkProps<TState = unknown>
+  extends React.PropsWithoutRef<LinkProps<TState>> {
+  /** A class to apply when this Link is active */
+  activeClassName?: string
+
+  /** Inline styles for when this Link is active */
+  activeStyle?: {}
```

Comment:
does react ship a style type that we can use for this?

Reply:
I couldn't find it, but I'm not sure where these come from in js. Do you know what the corresponding npm package that gives us these? Doesn't seem to be in react core